Added new sandcastle example for 2d/3d side by side synced camera views

OmarShehata opened the pull request:
This adds a new sandcastle example as described in #5749. The code is heavily commented so it serves as a learning example, explaining each step. 

I re-compute `viewCenter` on each sync to make sure it works with a window resize without having to make the code more verbose. 

@ggetz please review when you get a chance!

cesium-concierge commented:
![Signed CLA is on file.](https://img.shields.io/badge/CLA-signed-brightgreen.svg)

@OmarShehata, thanks for the pull request! Maintainers, we have a signed CLA from @OmarShehata, so you can review this at any time.

:warning: I noticed that [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CHANGES.md) has not been updated. If this change updates the public API in any way, fixes a bug, or makes any non-trivial update, please add a bullet point to `CHANGES.md` and comment on this pull request so we know it was updated. For more info, see the [Pull Request Guidelines]( https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#pull-request-guidelines).

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

ggetz commented:
Thanks @OmarShehata, I will review shortly. In the meantime, it looks like CI is failing on the `eslint` task. Try running the linting tool locally and fixing any errors.

ggetz commented on Apps/Sandcastle/gallery/Side by Side Camera.html:
```diff
@@ -0,0 +1,79 @@
+      }
+</style>
+<div id="Container" class="fullSize">
+    <div id="View3D"></div>
```
Our coding style for HTML id's is Camel Case, not Pascal Case. 

I don't think this or any HTML standards are mentioned in our Coding Guide. It would probably worth noting that we Camel Case HTML ids, and hyphenate CSS classes. Additionally, avoid inline `style` attributes, and prefer styling the class or id instead.

ggetz commented on Apps/Sandcastle/gallery/Side by Side Camera.html:
```diff
@@ -0,0 +1,79 @@
+    // The center of the view is the point that the 3D camera is focusing on
+    var viewCenter = new Cesium.Cartesian2(Math.floor(View3D.canvas.clientWidth / 2), Math.floor(View3D.canvas.clientHeight / 2));
+    // Given the pixel in the center, get the world position
+    var worldPos = View3D.scene.camera.pickEllipsoid(viewCenter);
```
No abbreviating variable names.

ggetz commented on Apps/Sandcastle/gallery/Side by Side Camera.html:
```diff
@@ -0,0 +1,79 @@
+// By default, the `camera.changed` event will trigger when the camera has changed by 50%
+// To make it more sensitive, we can bring down this sensitivity
+View3D.camera.percentageChanged = 0.01;
+// Setting it to 0 will cause it to trigger every frame and will be equivilent to running this in a `requestAnimationFrame`
```
We avoid using (or referencing) `requestAnimationFrame` because Cesium handles the render loop itself. Instead we'd recommend using the [`scene.postUpdate`](https://cesiumjs.org/Cesium/Build/Documentation/Scene.html?classFilter=scene#postRender) event (or another of the render loop related `scene` events).

OmarShehata commented:
I've renamed the file, made it sync across time, disabled camera controls on the 2D view, removed a bunch of widgets, and renamed the variables. Let me know if there's anything else I missed!

ggetz commented on Apps/Sandcastle/gallery/Multiple Synced Views.html:
```diff
@@ -0,0 +1,106 @@
+    // Given the pixel in the center, get the world position
+    var worldPosition = view3D.scene.camera.pickEllipsoid(viewCenter);
+    // Get the distance between the world position of the point the camera is focusing on, and the camera's world position
+    var distance = Cesium.Cartesian3.distance(worldPosition, view3D.scene.camera.positionWC);
```
When I tilt the camera view and the center of the camera is not on the ellipsoid, I get a developer error.

You'll probably want to check if world position is defined, and if not, just assume a base value for `distance`, or use a previous value.

ggetz replied:
Also to check if an object is defined, we use [`Cesium.defined`](https://cesiumjs.org/Cesium/Build/Documentation/defined.html).

ggetz commented on Apps/Sandcastle/gallery/Multiple Synced Views.html:
```diff
@@ -0,0 +1,106 @@
+//Sandcastle_Begin
+// We want our two views to be synced across time, so we create
+// a shared clock object that both views share
+var clockViewModel = new Cesium.ClockViewModel(new Cesium.Clock());
```
The `clock` parameter is optional -  https://cesiumjs.org/Cesium/Build/Documentation/ClockViewModel.html

Since we're not setting any specific configurations, I would omit it.

ggetz commented on Apps/Sandcastle/gallery/Multiple Synced Views.html:
```diff
@@ -0,0 +1,106 @@
+// a shared clock object that both views share
+var clockViewModel = new Cesium.ClockViewModel(new Cesium.Clock());
+var options3D = {
+    homeButton:false,
```
More code style things, the spacing of object properties are usuaully like so:

```
{
    homeButton : false,
    fullscreenButton : false
}

ggetz commented on Apps/Sandcastle/gallery/Multiple Synced Views.html:
```diff
@@ -0,0 +1,106 @@
+var view3D = new Cesium.Viewer('view3D', options3D);
+var view2D = new Cesium.Viewer('view2D', options2D);
+
+function sync2DView(){
```
Code styling spacing. For functions it's usually:
```
function sync2DView() {
```

OmarShehata commented:
I appreciate all your feedback @ggetz ! These issues should now be fixed. Travis seems to be failing with some timeouts. I ran `npm run test -- --browsers Electron --webgl-stub --failTaskOnError --suppressPassed` locally on `master` and those seem to be failing there as well. 

Any suggestions?

OmarShehata commented:
Thanks Gabby! And yeah, I can add it to my list, to find a place for HTML styles in our coding guide.

ggetz commented:
Thanks @OmarShehata!
